### ISSUE TITLE
Enhancement Request: Option to reference an existing secret for API Key or Bearer Token

### ISSUE THREAD

**Issue by asjonos** [OctopusDeploy/helm-charts#228]
It would be beneficial to add functionality allowing the referencing of an existing secret that holds an API key or a bearer token.

```
agent:
  existingApiKeySecret: my-secret-name
  existingBearerTokenSecret: my-secret-name
```

In the existing tentacle-deployment.yaml, we have:
```
            {{- if .Values.agent.serverApiKey }}
            - name: "ServerApiKey"
              valueFrom:
                secretKeyRef: 
                  name: {{ include "kubernetes-agent.secrets.serverAuth" . }}
                  key: api-key
            {{- end }}
            {{- if .Values.agent.bearerToken }}
            - name: "BearerToken"
              valueFrom:
                secretKeyRef: 
                  name: {{ include "kubernetes-agent.secrets.serverAuth" . }}
                  key: bearer-token
            {{- end }}
```
Add something to support the use of existing secrets:
```
            {{- if .Values.agent.existingApiKeySecret }}
            - name: "ServerApiKey"
              valueFrom:
                secretKeyRef: 
                  name: {{ .Values.agent.existingApiKeySecret }}
                  key: api-key
            {{- end }}
            {{- if .Values.agent.existingBearerTokenSecret }}
            - name: "BearerToken"
              valueFrom:
                secretKeyRef: 
                  name: {{ .Values.agent.existingBearerTokenSecret }}
                  key: bearer-token
            {{- end }}
```
Additionally, the tentacle-server-auth-secret.yaml must only be created if neither existingApiKeySecret nor existingBearerTokenSecret is set.

**Comment by APErebus** [comment]
Hi @asjonos!

Thanks for the feedback. I'll add this to our backlog

**Comment by APErebus** [comment]
Hi @asjonos 

I had quick play with this today and it looks like it was pretty easy. See #245.

I'll look at getting it reviewed and merged soon

**Comment by APErebus** [comment]
Hi @asjonos 

This has been implemented and is/will be available in `1.13.0` of the Kubernetes agent helm chart.

A key thing to note is that the namespace and secrets must exist before the agent is installed. The comments in the values files let you know what data nodes the values need to exist at
https://github.com/OctopusDeploy/helm-charts/blob/07503d105fc6e8082dc6af41b0990935fc5fe259/charts/kubernetes-agent/values.yaml#L37-L54
